lib/wasi/src/syscalls/mod.rs

```diff
@@ -2515,18 +2516,19 @@ pub fn proc_raise(env: &WasiEnv, sig: __wasi_signal_t) -> __wasi_errno_t {
 ///     A pointer to a buffer where the random bytes will be written
 /// - `size_t buf_len`
 ///     The number of bytes that will be written
-pub fn random_get(env: &WasiEnv, buf: WasmPtr<u8, Array>, buf_len: u32) -> __wasi_errno_t {
+pub fn random_get(env: &WasiEnv, buf: u32, buf_len: u32) -> __wasi_errno_t {
     debug!("wasi::random_get buf_len: {}", buf_len);
     let memory = env.memory();
-
-    let buf = wasi_try!(buf.deref(memory, 0, buf_len));
-
-    let res = unsafe {
-        let u8_buffer = &mut *(buf as *const [_] as *mut [_] as *mut [u8]);
-        getrandom::getrandom(u8_buffer)
-    };
+    let mut u8_buffer = vec![0; buf_len as usize];
+    let res = getrandom::getrandom(&mut u8_buffer);
     match res {
-        Ok(()) => __WASI_ESUCCESS,
+        Ok(()) => {
+            memory
+                .view::<u8>()
+                .subarray(buf, buf + buf_len)
```

Comment:
Same here, we will now panic given inputs that we previously handled correctly

Reply:
Fixed. I transformed them to `usize`